src/lib/form-field/form-field.ts

```diff
@@ -468,12 +471,13 @@ export class MatFormField extends _MatFormFieldMixinBase
 
       const containerStart = this._getStartEnd(
           this._connectionContainerRef.nativeElement.getBoundingClientRect());
-      const labelStart = this._getStartEnd(
-          this._label.nativeElement.children[0].getBoundingClientRect());
+      const labelStart = this._getStartEnd(labelEl.children[0].getBoundingClientRect());
       let labelWidth = 0;
-      for (const child of this._label.nativeElement.children) {
+
+      for (const child of labelEl.children) {
         labelWidth += child.offsetWidth;
       }
+
       this._outlineGapStart = labelStart - containerStart - outlineGapPadding;
       this._outlineGapWidth = labelWidth * floatingLabelScale + outlineGapPadding * 2;
```

Comment:
should probably also add logic to make the `_outlineGapWidth` `0` if the `labelWidth` is `0`

Reply:
Done.